src/socket.c

```diff
@@ -623,28 +663,3 @@ int trilogy_sock_upgrade_ssl(trilogy_sock_t *_sock)
     sock->ssl = NULL;
     return TRILOGY_OPENSSL_ERR;
 }
-
-int trilogy_sock_discard(trilogy_sock_t *_sock)
```

Comment:
I fear the `dup2` dance may still be necessary.

If the socket wasn't fully flushed, calling `close()` on it may send a `RST` packet to the server. 

So in a forking situation, if that happens in the child (because you discard your inherited connections), then the parent connection will be reset.

Of course all this is a bit handwavy, and if the connection was properly flushed it shouldn't happen, but...

Reply:
I don't think that's the case. If the socket is still open in the parent process, the `close` in the child should just be decrementing a counter in the kernel. The kernel will only send a `RST` or `FIN` when the last copy is closed.

As far as I can see `dup2` is equivalent to calling `close` on the second argument.

macOS says:
> if descriptor fildes2 is already in use, it is first deallocated as if a close(2) call had been done first.

Linux man-pages 6.04 says:
> If  the file descriptor newfd was previously open, it is closed before being reused; the close is performed silently (i.e., any errors during the close are not re‐ported by dup2()).
>
> The steps of closing and reusing the file descriptor newfd are performed atomically.  This is important, because trying to implement equivalent functionality using close(2) and dup() would be subject to race conditions, whereby newfd might be reused between the two steps.

I think calling close has identical behaviour (from the remote's perspective) to what we did previously.

Reply:
Alright, I think you are right. 👍